addons/mail/static/src/utils/common/format.js

```diff
@@ -207,6 +210,13 @@ function generateMentionsLinks(body, { partners = [], threads = [], specialMenti
             markup(`<a href="#" class="o-discuss-mention">@${htmlEscape(special)}</a>`)
         );
     }
+    for (const role of roles) {
+        body = htmlReplace(
+            body,
+            `@${role.name}`,
+            markup(`<a href="#" class="o-discuss-mention">@${htmlEscape(role.name)}</a>`)
```

Comment:
We might want to have the actual role page as URL rather # to allow proper following of the link.

Reply:
We might want to build it in a similar way as `mentions` below.